### PR TITLE
feat: whitelist support function && English README

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- `whitelist` support function `(extname, filename) => true`
- **English ReadMe**
- code style adjust
- close https://github.com/eggjs/egg/issues/774

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,19 +20,18 @@
 [download-image]: https://img.shields.io/npm/dm/egg-multipart.svg?style=flat-square
 [download-url]: https://npmjs.org/package/egg-multipart
 
-通过内置 [multipart](https://github.com/cojs/multipart) 实现了文件流式上传，
-能做到请求文件数据不落地本地磁盘即可完成处理。
+Use [multipart](https://github.com/cojs/multipart) to upload file by streaming and process it without save to disk.
 
-在应用中通过 `this.multipart()` 拿到文件流，直接传给图片处理模块 gm 或者直接上传到云存储 tfs 和 oss 都是可以实现的。
+Just use `ctx.multipart()` to got file stream, then pass to image processing liberary such as `gm` or upload to cloud storage such as `tfs` or `oss`.
```
Comment (dead-horse) [inline README.md]: tfs 去掉

File: README.md [hunk 2]
```diff
@@ -59,93 +58,102 @@ var whitelist = [
...
 };
 ```
 
-__注意，当传递了 `whitelist` 参数的时候 `fileExtensions` 参数失效。__
+Or by function：
 
-## 使用示例
+```js
+exports.multipart = {
+  whitelist: (extname, filename) => [ '.png' ].includes(extname)
```
Comment (dead-horse) [inline README.md]: 直接给 filename 吧，extname 包含在 filename 里面了

File: README.md [hunk 3]
```diff
@@ -20,19 +20,18 @@
 [download-image]: https://img.shields.io/npm/dm/egg-multipart.svg?style=flat-square
 [download-url]: https://npmjs.org/package/egg-multipart
 
-通过内置 [multipart](https://github.com/cojs/multipart) 实现了文件流式上传，
-能做到请求文件数据不落地本地磁盘即可完成处理。
+Use [multipart](https://github.com/cojs/multipart) to upload file by streaming and process it without save to disk.
```
Comment (atian25) [inline README.md]: @fengmk2 @dead-horse 这里是不是改过了？没看到有引入这个库

File: README.md (hunk 2 above)
Comment (atian25) [inline README.md]: done

**Comment by atian25** [comment]
再看看这个

**Comment by atian25** [comment]
ping

File: README.md (hunk 3 above)
Comment (dead-horse) [inline README.md]: 没有引它了，直接用了 co-busboy 吧

**Comment by dead-horse** [comment]
others LGTM

File: README.md (hunk 3 above)
Comment (atian25) [inline README.md]: 改了

**Comment by dead-horse** [comment]
1.3.0

**Pull request merged**